crates/abi/src/function.rs

```diff
@@ -31,6 +31,8 @@ impl AbiFunction {
         name: String,
         args: Vec<(String, AbiType)>,
         ret_ty: Option<AbiType>,
+        has_self: bool,
+        has_ctx: bool,
```

Comment:
It's better to take simply `state_mutability` instead of introducing Fe-specific concepts like `self` and `ctx` to avoid unnecessary tight coupling with Fe language specification.

Reply:
> `mut` would be launched soon. So I think it'd be better to finish the PR after its deployment.

Okay. I got it!